File: Java/Donaciones/src/donaciones/Administrador.java
package donaciones;
import java.util.ArrayList;
import java.util.List;

/**
 *
 * @author mario21ic
 */
public class Administrador {
    private List<Donante> donantes = new ArrayList<Donante>();
    
    public void registrarDonante(Donante donante){
        this.donantes.add(donante);
    }
    
    public double montoTotal() {
        double total = 0.00;
        for(Donante d:donantes) {
            total += d.getMonto();
        }
        return total;
    }
    
    public String donanteTop() {
        String donanteTop = "";
        double montoAnterior = 0.00;
        for(Donante d:donantes) {
            if (d.getMonto() > montoAnterior) {
                donanteTop = d.getNombre();
            }
            montoAnterior = d.getMonto();
        }
        return donanteTop;
    }
    
    public Donante buscarDonante(String dni) {
        for(Donante d:donantes) {
            if (d.getDni() == dni) {
                return d;
            }
        }
        Donante donante = new Donante();
        return donante;
    }
    
    public double montoPromedio() {
        return (this.montoTotal() / donantes.size());
    }
}

File: PHP/Donaciones/Administrador.php
<?php

class Administrador {
    private $_donantes = array();

    public function registrar_donante($donante) {
        $this->_donantes[] = $donante;
    }

    public function monto_total() {
        $total = 0.00;
        foreach ($this->_donantes as $donante) {
            $total += $donante->getMonto();
        }
        return $total;
    }

    public function donante_top() {
        $monto_anterior = 0.00;
        $donante_top = "";
        foreach($this->_donantes as $donante) {
            if ($donante->getMonto() > $monto_anterior) {
                $donante_top = $donante->getNombre();
            }
            $monto_anterior = $donante->getMonto();
        }
        return $donante_top;
    }

    public function buscar_donante($dni) {
        foreach($this->_donantes as $donante) {
            if ($donante->getDni() == $dni) {
                return $donante;
            }
        }
        return FALSE;
    }

    public function monto_promedio() {
        return ($this->monto_total() / count($this->_donantes));
    }
}
